gnat2goto/driver/tree_walk.adb

```diff
@@ -1235,6 +1306,14 @@ package body Tree_Walk is
                when Attribute_Last   =>
                   return Report_Unhandled_Node_Irep (N, "Do_Expression",
                                                      "Last attribute");
+               when Attribute_Val =>
+                  return Do_Attribute_Pos_Val (N);
+               when Attribute_Pos =>
+                  return Do_Attribute_Pos_Val (N);
+               when Attribute_Pred =>
```

Comment:
could we not have a check for floating point type here and fail with a warning message?